conda_forge_tick/migrators/migration_yaml.py

```diff
@@ -146,8 +146,7 @@ def __init__(
             ignored_deps_per_node=ignored_deps_per_node,
         )
         self.yaml_contents = yaml_contents
-        assert isinstance(name, str)
```

Comment:
Why was this assert removed?

Reply:
The type annotation of `name` is `Optional[str]`, so it doesn't make a lot of sense to assert a string type IMO.
On top of that, I'd suggest sticking to type annotations for hinting types to developers.
But thanks for pointing out, I think actually the correct solution is to change the annotation to `str` - the name ultimately comes from `os.path.splittext` which returns a string. I will change that - but I don't see why we need the assertion then.

Reply:
fixed in 4aecb95e

Reply:
I don't recall when or why the assert was added, but I assume it was for a reason. Unless we can figure that out and ensure it is safe to remove, we should keep it.

Reply:
Moved to #2273.